### DISCUSSION TITLE
Intended use-case

### DISCUSSION THREAD

**Discussion by Yakimych** [psfinaki/Finelines#3] [Q&A] [answered]
Just wondering about a practical use-case in a real project. If I understand correctly, the idea is to:
- Write some F# code for describing a pipelines
- Build and run the code locally, save output to a yaml file
- Commit the changes and the yaml file to source control

This differs from a possible use-case for [farmer](https://github.com/CompositionalIT/farmer), where one can build a Farmer project as part of the pipeline itself (without committing the generated ARM template to source control). Am I correct in that such a use-case is not possible for generating a YAML file, as one would already need a YAML file to even trigger the build/perform a build step?

**Comment by psfinaki** [comment, marked as answer]
Hi @Yakimych,

Yes, all you've written is correct. I was thinking about this a bit and yes I welcome the ideas on how to integrate this better in the CI/CD. Indeed, at least in Azure DevOps one already needs a YAML pipeline to do anything.

Like, one way could be to have a single step to generate a real pipeline from the code and then execute it on the fly. Actually I wonder if even this is possible - executing generated YAML on the fly. But anyway, it's not the most elegant way to go...

Or maybe you have some fresh thoughts on that?

**Reply by Yakimych** [reply]
Understood, thanks for the clarification, @psfinaki!

I am not an expert on Azure Pipelines, but it does not seem to be possible to generate and execute YAML on the fly: https://github.com/microsoft/azure-pipelines-yaml/issues/222#issuecomment-511526898. The feature request for this has been closed a white back 🙁

**Reply by psfinaki** [reply]
Yep, that's sad. I've [asked](https://stackoverflow.com/questions/68695935/is-there-any-way-to-simulate-dynamic-yaml-pipelines-in-azure-devops) on StackOverflow to be sure about it, maybe somebody has extra ideas...

**Reply by psfinaki** [reply]
Okay doesn't look like extra ideas. Well, maybe that's life then, at least for ADO. Probably other CI/CD have features like this, I will look into GitHub Actions soon.
